### PR TITLE
[agent-a][issue #564] Promote historical replay owner

### PR DESCRIPTION
**Pull request by yazzaoui**
## Summary

Part of #564. Related to #549 and blocked sibling #629.

Promotes `runtime.run_fork.historical_replay_execution` from a narrow delivery-event replay wrapper into the canonical mutating authority boundary for timestamp-fork historical replay execution decisions after admission.

This PR does not claim product-complete replay/resume for every future fact family. Its closure claim is canonical owner promotion: downstream store/runtime mutation now consumes owner-authorized executable work from the runtime historical replay owner, and unsupported families remain explicit blockers/splits.

## Governing refs/context

- `docs/specs/swarm-platform/platform/contracts/platform-spec.yaml`: fork sections `activation_admission`, `historical_replay_execution_admission`, new/updated `historical_replay_execution`, `historical_replay_delivery_event_replay_execution`, semantics `3b`, `3j`, `3j1`, `3k`, and `3k1`.
- #564 Broad Refactor Pre-Implementation Coverage Audit and independent gate outcome: approved for broad refactor / canonical owner promotion only.
- #631/#632: prerequisite non-mutating historical replay execution admission.
- #633/#634: prerequisite limited delivery-event replay mutation.
- #635/#636: prerequisite limited delivery-event replay recovery.
- #629: blocked contract-swap sibling proving narrow mutation cannot proceed without the broad owner.
- `docs/IMPLEMENTER_GUIDELINES.md` and `docs/SEMANTIC_DRIFT.md`.

## Semantic migration

- New canonical mutating authority boundary: `runtime.run_fork.historical_replay_execution`.
- Old producer/reader/writer path now invalid: store activation and `store.run_fork.delivery_event_replay` selecting executable delivery-event replay work directly from `RunForkPlan`, `ReplayResumeAdmission`, or source delivery rows.
- Production paths checked for surviving old behavior: activation, historical replay execution builder, store delivery-event replay writer, delivery-event lineage writer, committed replay-scope marker writer, startup/sweeper recovery consumers, platform spec, and runtime watchlist.
- Migration completeness: complete for #564's approved canonical owner promotion class. Full product replay/resume remains open for timers, routes, sessions, turns, audits, non-agent replay, contract-swap full boot/resume, broader source-advanced policy, restart recovery beyond the safe primitive, and API/dashboard/Builder consumers.

## Tests

- `go test ./internal/runtime/runforkexecution ./internal/store -run 'HistoricalReplayExecution|DeliveryEventReplay|ActivateRunFork' -count=1`
- `go test ./...`
- `python3` YAML parse for `platform-spec.yaml` and `runtime-operations.yaml`
- `git diff --check`

### CODE REVIEW

**Comment by yazzaoui** [comment]
## Post-Implementation Proof Audit

### Semantic Concepts Changed

- `runtime.run_fork.historical_replay_execution` as the canonical mutating authority for timestamp-fork historical replay execution decisions after admission.
- Delivery-event replay mutation now consumes owner-authorized executable work identities emitted by that runtime owner.
- `store.run_fork.delivery_event_replay` is explicitly a physical fork-local row writer, not the semantic owner that selects replayable source deliveries.
- The full historical source fact matrix remains explicit as executable, reconstructed, lineage-only, fail-closed, or split sibling.

Chosen working failure class closed by this PR: `runtime.run_fork.historical_replay_execution.full_mutating_replay_resume_owner` at the approved closure level of canonical owner promotion with only `delivery_event_replay_ready` executable.

Immediate parent failure class: #564 full timestamp-fork historical replay/resume.

Parent above it: #549 run-isolated fork architecture/current-state parent.

### Canonical Owners After Change

| Concept | Canonical owner after this PR | Touched consumers |
| --- | --- | --- |
| Historical replay execution mutation authority | `runtime.run_fork.historical_replay_execution` | `BuildHistoricalReplayExecution`, `ActivateRunFork`, `store.run_fork.delivery_event_replay`, tests, spec, watchlist |
| Non-mutating historical replay fact matrix | `runtime.run_fork.historical_replay_execution_admission` | runtime execution builder, activation, tests, spec |
| Replay/resume source taxonomy | `store.run_fork.replay_resume_admission` | runtime owner consumes; store activation no longer uses it as direct mutation authorization |
| Physical fork-local delivery-event replay row writes | `store.run_fork.delivery_event_replay` | activation passes owner-authorized work; writer validates runtime owner before writes |
| Recovery for safe replay primitive | `runtime.run_fork.historical_replay_execution.delivery_event_replay_ready_recovery` | unchanged; recovery still consumes fork-local rows through persisted replay owners |

### Exhaustive Systematic-Consumption Audit

| Seam | Status after PR |
| --- | --- |
| `BuildHistoricalReplayExecution` | moved to broad owner: validates complete fact matrix, rejects unsupported executable families, emits `DeliveryEventReplayWork` identities. |
| `ActivateRunFork` | moved to canonical owner: requires `RunForkHistoricalReplayExecution` when taxonomy says delivery replay is ready, and passes owner-authorized work to the writer. |
| `store.run_fork.delivery_event_replay` | moved to canonical owner: refuses writes unless `runtime.run_fork.historical_replay_execution` authorized executable `event_deliveries`. |
| `store.run_fork.replay_resume_admission` | prerequisite only: keeps taxonomy predicate; does not directly authorize row writes. |
| selected-contract execution | different concept: remains supported selected frontier executor, not full historical resume. |
| contract-swap admission / #629 | still split/blocked: non-mutating readiness only; not mutation authorization. |
| receipts, dead letters, retry/idempotency, follow-ups | explicit split/blocker outputs; fork lifecycle only for admitted fork-local work. |
| timers, routes, sessions, turns, audits, non-agent work | explicit split/fail-closed outputs; no helper ownership introduced. |
| source post-T facts/outcomes | lineage/proof only or blocker; no suppressor path introduced. |
| RecoveryManager / sweeper | unchanged consumer of fork-local rows; source rows remain non-authoritative. |
| CLI/API/dashboard/Builder | consumers only; no semantic ownership added. |

Old paths now invalid / non-authoritative / still surviving:

- Invalid/non-authoritative: activation or delivery-event writer selecting replayable work directly from `RunForkPlan.PendingWork`, `ReplayResumeAdmission.DeliveryEventReplayReady`, or source rows.
- Invalid/non-authoritative: source event/delivery/receipt/dead-letter/retry/outcome rows as executable fork truth or suppressors.
- Still surviving by design: `store.run_fork.replay_resume_admission` as prerequisite taxonomy; `store.run_fork.delivery_event_replay` as physical writer; selected execution as selected frontier executor; recovery owners for fork-local rows.

### Broader Failure Class And Closure Level

Broader failure class: full product historical replay/resume under #564/#549.

Issue was symptom-shaped: yes. The observed tail after #636 was shaped by remaining sibling manifestations, but the approved boundary was the broad canonical owner promotion rather than one local endpoint.

Achieved closure level: canonical owner promotion / historical replay execution ownership drift eliminated for the currently executable primitive. This PR does not claim product-complete replay/resume for every unsupported future fact family.

Parent class remains open after this PR: yes. #564/#549 remain open for timers, routes, sessions, turns, audits, non-agent replay, contract-swap full boot/resume, broader source-advanced policy, restart recovery beyond the safe primitive, and API/dashboard/Builder consumers.

Remaining child-slice tail estimate: unchanged from the pre-audit except that owner promotion is now complete. Remaining groups are fact-family expansion children, not local bypass repair. Confidence: high.

### Sibling Contexts Checked

Checked and kept split or different concept: selected-contract execution, contract-swap admission/#629, selected route recovery/#618, delivery-event replay recovery/#635, same-run committed replay scope, receipts/dead letters/retry/idempotency, follow-ups, timers, routes, sessions, turns, audits, non-agent/node/system work, source post-T facts, and CLI/API/dashboard/Builder.

Generic failing proof used/created:

- Runtime unit proof that `BuildHistoricalReplayExecution` rejects taxonomy-ready input without owner-authorized replay work.
- Store/runtime integration proof that activation passes pending work into the owner and the writer consumes the owner work result.
- Spec/watchlist proof that source row copy and direct plan/admission replay selection are invalid.

Watchlist decision: existing node refined. `docs/watchlists/runtime-operations.yaml` now records broad historical replay execution owner promotion as the active manifestation and adds #629 to representative issue evidence.

Architecture feedback: the long-run issue remains product-complete fact-family replay/resume expansion, but the type-model smell of distributed mutation authority is closed for the current primitive by requiring executable work identities from the runtime owner.

Tracking decision: #564/#549 remain the open parent trackers; #629 remains blocked until contract-swap full resume consumes the broad owner. No new child issue was opened in this PR.

### Manifestation Coverage Table

| Manifestation | Status | Exact proof |
| --- | --- | --- |
| Activation/store could previously pass `RunForkPlan` into delivery-event replay writer | reproduced and fixed | `ActivateRunFork` now passes `RunForkHistoricalReplayExecution`; `go test ./internal/store -run ActivateRunFork -count=1` via focused command. |
| Delivery-event row writer could select replayable work from plan pending rows | reproduced and fixed | `applyRunForkDeliveryEventReplay` now requires `runtime.run_fork.historical_replay_execution` and consumes `DeliveryEventReplayWork`; focused store tests passed. |
| Runtime historical owner did not carry the complete fact matrix into mutation result | reproduced and fixed | `RunForkHistoricalReplayExecution` now includes `FactAdmissions`, required consumers, closure level, and full-replay support flag; runtime tests assert this. |
| Runtime historical owner could construct execution without concrete owner-selected work | reproduced and fixed | `TestBuildHistoricalReplayExecutionRejectsTaxonomyReadyWithoutOwnerWork`. |
| Source events | execution-proven through corrected path | Spec and execution model keep source events lineage only; writer mints deterministic fork-local event IDs. |
| Source deliveries | execution-proven through corrected path | Only owner-emitted `DeliveryEventReplayWork` identities are writable; source delivery rows are lineage inputs. |
| Fork-local events/deliveries | reproduced and fixed | Existing activation/recovery tests prove fork-local rows and lineage; writer now validates owner. |
| Receipts | split / escalated as separate class | Fact matrix remains lineage/split; source receipts cannot suppress fork work. |
| Dead letters | split / escalated as separate class | Fact matrix remains lineage/fail-closed; no dead-letter replay added. |
| Retry/idempotency | split / escalated as separate class | Fact matrix remains split; no source retry/idempotency suppressor added. |
| Emitted follow-ups | split / escalated as separate class | Fact matrix remains split; follow-ups regenerate only from admitted fork-local execution. |
| Timers | split / escalated as separate class | Fact matrix/spec keep timer reconstruction split/fail-closed. |
| Routes | split / escalated as separate class | Selected route recovery remains separate; broad historical route reconstruction not added. |
| Sessions | split / escalated as separate class | #573 policy remains; no session reconstruction added. |
| Turns | split / escalated as separate class | #573 policy remains; no turn reconstruction added. |
| Audits | split / escalated as separate class | #573 policy remains; no audit reconstruction added. |
| Non-agent/node/system work | split / escalated as separate class | #575 policy remains; no handler replay added. |
| Source-advanced/post-T outcomes | execution-proven as non-suppressors | Spec/watchlist state source outcomes are lineage/proof only; no suppressor path introduced. |
| Contract-swap full boot/resume | split / escalated as separate class | #629 remains blocked; spec says broad owner is required first. |
| Restart recovery | execution-proven through existing corrected path | #635 recovery continues to consume fork-local rows; full recovery remains split. |
| CLI/API/dashboard/Builder | split / escalated as consumer-only | No surface ownership added; spec/watchlist keep consumers only. |

### Required Supported-Surface Proof Actually Run

This is not a parity issue and no public CLI/API output changed. Required proof was runtime/store plus spec/watchlist validation.

Proof run:

- `go test ./internal/runtime/runforkexecution ./internal/store -run 'HistoricalReplayExecution|DeliveryEventReplay|ActivateRunFork' -count=1`
- `go test ./...`
- `python3` YAML parse for `platform-spec.yaml` and `runtime-operations.yaml`
- `git diff --check`

File: internal/store/run_fork_delivery_event_replay.go [hunk 1]
```diff
@@ -37,21 +37,32 @@ type runForkReplaySourceEvent struct {
...
 	}
-	replayable := runForkReplayablePendingWork(plan.PendingWork)
+	if strings.TrimSpace(execution.Owner) != RunForkHistoricalReplayExecutionOwner ||
+		strings.TrimSpace(execution.AdmissionOwner) != RunForkHistoricalReplayExecutionAdmissionOwner ||
+		!execution.DeliveryEventReplayReady ||
+		execution.EventDeliveriesAdmission.Fact != RunForkHistoricalReplayFactEventDeliveries ||
+		execution.EventDeliveriesAdmission.Admission != RunForkHistoricalReplayAdmissionExecutableForkWork {
+		return result, fmt.Errorf("store.run_fork.delivery_event_replay requires %s owner-authorized executable event_deliveries", RunForkHistoricalReplayExecutionOwner)
+	}
+	replayable := execution.DeliveryEventReplayWork
```
Comment (yazzaoui) [inline internal/store/run_fork_delivery_event_replay.go:53]: This consumes owner-emitted work, but the store never proves that each emitted `SourceDeliveryID` is still part of the current `RunForkPlan.PendingWork` safe subset. `loadRunForkReplaySourceEvent` only verifies the source event; the lineage FK only proves the delivery exists somewhere. A stale/malformed owner output can therefore replay a delivered/in-progress/non-agent/foreign delivery while carrying valid owner strings. Please fail closed by validating each work item against the current plan evidence before mutation, including event/run membership, pending/unstarted agent shape, subscriber identity, classification/status/retry/session/timestamps, and duplicate rejection.

**Review by yazzaoui — COMMENTED** [review]
Changes needed.

The broad failure-class boundary is acceptable for the approved #564 broad-refactor slice: this PR promotes `runtime.run_fork.historical_replay_execution` as the canonical mutating authority for the currently supported `delivery_event_replay_ready` subset, not product-complete historical replay/resume. The pre-audit gate exists and was independently approved for that exact closure level, and the spec/watchlist updates are present.

Blocking issue: the store-side mutation path consumes `RunForkHistoricalReplayExecution.DeliveryEventReplayWork`, but it does not validate that each owner-emitted work identity is still a member of the just-revalidated `RunForkPlan.PendingWork` safe subset. It checks owner strings, admission labels, non-empty IDs, and source event existence, but not that `SourceDeliveryID` belongs to the source run/source event, is pending/unstarted agent work, or matches the subscriber/classification from the canonical replay taxonomy. Because `ActivateRunFork` accepts an injected `RunForkHistoricalReplayExecutionAdmitter`, a malformed or stale owner output can authorize a fork-local pending delivery/lineage row for a delivered, in-progress, non-agent, or foreign source delivery as long as the referenced delivery exists. That weakens the fail-closed boundary this PR is trying to establish.

Required fix: before any delivery-event replay mutation, validate owner-emitted work as an exact subset of the current plan evidence: match delivery_id, event_id, subscriber_type, subscriber_id, classification/status/retry/session/timestamps using the canonical replayable predicate, reject duplicates, reject foreign/mismatched deliveries, and add negative tests proving malformed owner output fails before source freeze, fork activation, fork event creation, delivery creation, and lineage insertion.

Note: GitHub rejected a formal REQUEST_CHANGES action because the authenticated user owns this PR; this review comment is intentionally blocking.

**Comment by yazzaoui** [comment]
Checklist review for #637

- Guideline checks: pre-audit gate present and independently recorded on #564; cited spec sections inspected; PR proof audit present; parent/sibling tracker state checked; watchlist update present.
- Failure-class assessment: chosen class is broad enough for the approved owner-promotion slice, not product-complete #564 historical replay/resume. #549 remains open, #564 remains open, #629 remains blocked, #631/#633/#635 are closed, #618 is closed.
- Spec compliance: incomplete until the store validates owner-emitted executable work identities against the current plan evidence before mutation. The spec requires only pending unstarted source-agent deliveries admitted as `delivery_event_replay_ready`, with source rows as lineage/evidence only.
- Risk level: high, because the remaining gap is a fail-closed semantic boundary on a mutating runtime/store path.
- Closure level if fixed: canonical owner promotion with `delivery_event_replay_ready` only. Not product-complete historical replay/resume.
- Residual risk after fix: timers, routes, sessions, turns, audits, non-agent/node/system replay, contract swap/full resume, broader restart recovery, source-advanced policy, and API/dashboard/Builder remain split under #564/#549/#629.
- Watchlist decision: no additional watchlist change required; the learned risk is already under `timestamp_fork_replay_resume_ownership` / delivery-event replay owner-consumption bullets added in this PR.
- Merge recommendation: do not merge yet. Changes needed: validate each `DeliveryEventReplayWork` item as an exact safe subset of current `RunForkPlan.PendingWork` before any source freeze/fork activation/fork-local event or delivery write/lineage insert; add negative tests for stale, foreign, delivered/in-progress, non-agent, and duplicate owner work.

File: internal/store/run_fork_delivery_event_replay.go (hunk 1 above)
Comment (yazzaoui) [reply internal/store/run_fork_delivery_event_replay.go:53]: Addressed in `dab1da2c`.

Changes made:
- `ActivateRunFork` now validates `RunForkHistoricalReplayExecution.DeliveryEventReplayWork` against the current `RunForkPlan.PendingWork` before source freeze or any fork-local replay mutation.
- The validator rejects owner work that is missing from current evidence, duplicated, not replayable pending-agent work, or mismatched on event/subscriber/classification/reason evidence.
- Added negative coverage for stale/missing, foreign, delivered, duplicate, subscriber mismatch, plus direct unsafe-evidence coverage for in-progress, non-agent, and retry cases.
- The activation negative tests assert no source freeze, fork activation, fork event/delivery creation, or lineage insert occurred.

Proof run locally:
- `go test ./internal/store -run 'RunForkActivation_(ReplaysSafePendingDeliveryWithForkLocalLineage|RejectsOwnerWorkOutsideCurrentSafePendingEvidence)|RunForkDeliveryEventReplayValidationRejectsUnsafeCurrentEvidence' -count=1`
- `go test ./internal/runtime/runforkexecution ./internal/store -run 'HistoricalReplayExecution|DeliveryEventReplay|ActivateRunFork' -count=1`
- `go test ./...`
- `git diff --check`

**Review by yazzaoui — COMMENTED** [review]
Approved from process re-review; GitHub rejected formal APPROVE because this identity owns the PR.

The prior blocking store-side fail-closed gap is fixed in `dab1da2c`. `ActivateRunFork` now validates `RunForkHistoricalReplayExecution.DeliveryEventReplayWork` against the current `RunForkPlan.PendingWork` before source freeze and before any fork-local replay mutation. The validator rejects owner work that is missing from current evidence, duplicated, not replayable pending-agent work, or mismatched on event/subscriber/classification/reason evidence. The store writer remains a physical writer consuming the runtime historical replay execution owner rather than selecting replayable deliveries directly from plan/admission/source rows.

I resolved the prior inline thread after verifying the patch and tests. No blocking findings remain for the approved #564 broad-refactor closure level: canonical owner promotion with `delivery_event_replay_ready` only, not product-complete historical replay/resume.

**Comment by yazzaoui** [comment]
Checklist re-review for #637 at `dab1da2c5af0cd51a446dd044bdc8db1d26845b7`

- Guideline checks: PR description re-read; PR comments and review threads re-read; #564 issue/thread and independent broad-refactor gate checked; linked parent/sibling state checked for #549/#564/#629; cited spec/watchlist sections inspected through the changed files and YAML parse.
- Prior blocker: fixed. `ActivateRunFork` validates `RunForkHistoricalReplayExecution.DeliveryEventReplayWork` against current `RunForkPlan.PendingWork` before source freeze and before fork-local replay mutation.
- Spec compliance: acceptable for the approved closure level. Store delivery-event replay consumes runtime historical replay execution owner work and no longer relies on direct plan/admission/source-row replay selection as mutation authority.
- Failure-class coverage: broad enough for #564 canonical owner promotion with `delivery_event_replay_ready` only. This remains explicitly not product-complete historical replay/resume.
- Concept-wide drift assessment: no live same-concept store bypass found for the current primitive. Unsupported fact families remain explicit split/fail-closed siblings under #564/#549/#629.
- Closure level: canonical historical replay execution owner promotion achieved for the currently executable delivery-event replay primitive.
- Risk level: medium. The fixed path is mutating runtime/store code, but the prior high-risk gap now has negative coverage and full suite proof.
- Tests/proofs run by reviewer: `go test ./internal/store -run 'RunForkActivation_(ReplaysSafePendingDeliveryWithForkLocalLineage|RejectsOwnerWorkOutsideCurrentSafePendingEvidence)|RunForkDeliveryEventReplayValidationRejectsUnsafeCurrentEvidence' -count=1`; `go test ./internal/runtime/runforkexecution ./internal/store -run 'HistoricalReplayExecution|DeliveryEventReplay|ActivateRunFork' -count=1`; YAML parse for `docs/specs/swarm-platform/platform/contracts/platform-spec.yaml` and `docs/watchlists/runtime-operations.yaml`; `git diff --check origin/master...HEAD`; `go test ./...`.
- Residual risk: timers, routes, sessions, turns, audits, non-agent/node/system replay, contract-swap full boot/resume, broader source-advanced policy, restart recovery beyond the safe primitive, and API/dashboard/Builder remain split under the existing parent trackers.
- Follow-up/tracker decision: no new issue required. #564 and #549 remain open parent trackers; #629 remains the blocked contract-swap sibling.
- Watchlist decision: no additional watchlist edit required beyond this PR's runtime-operations update.
- Merge recommendation: mergeable from process review. GitHub rejected formal APPROVE because this identity owns the PR, so approval is recorded as a COMMENT review.
